flink-libraries/flink-table/src/main/scala/org/apache/flink/table/calcite/FlinkTypeFactory.scala

```diff
@@ -269,6 +272,10 @@ class FlinkTypeFactory(typeSystem: RelDataTypeSystem) extends JavaTypeFactoryImp
 
     canonize(newType)
   }
+
+  override def getDefaultCharset: Charset = {
+    Charset.forName(ConversionUtil.NATIVE_UTF16_CHARSET_NAME)
```

Comment:
Should we do UTF-8 by default?

You can just use `java.nio.charset.StandardCharsets.UTF_8` to avoid introducing additional dependency.

Reply:
+1 doesn't having UTF-8 everywhere solve all problems?

Reply:
Calcite only supports UTF-16, I guess because this is the default of Java Strings anyway.